### ISSUE TITLE
[question] [need for help]import a js file but does not has any export

### ISSUE THREAD

**Issue by whatwg6** [import-js/eslint-plugin-import#2402]
Encounter a problems.

I was using `eslint-webpack-plugin` in my project. Forget export toString in file1.js, but import toString In file2.js and toString was undefined.

```js
// file1.js
const toString = (p) => String(p)
```

```js
// file2.js
import a, { toString } from "file1.js"

console.log(a) // {}
console.log(toString) // undefined
```

I expect  report warn or error in file2.js. Does any rule can find this problems？

**Comment by ljharb** [comment]
Yes, the “named” rule should do that.

The recommended and Airbnb config both should enable this rule for you.

**Comment by whatwg6** [comment]
@ljharb hello, i was use import/named rule before ask this question. 

situation 1 will report import toString was not found

```js
// file1.js
const toString = (p) => String(p)
const toNumber = (p) => Number(p)

export {
 toNumber
}
```
```
// file2.js
import { toString } from "file1.js" // export 'default' (imported as 'xx') was not found in './xxx' (possible exports: xxx)
```

situation 2 does not has any report

```js
// file1.js
const toString = (p) => String(p)

```
```
// file2.js
import { toString } from "file1.js"
console.log(toString) // undefined
```

**Comment by ljharb** [comment]
Have you set the rule’s commonjs option to true?

**Comment by whatwg6** [comment]
@ljharb 

It was reproduce https://github.com/whatwg6/eslint-import-named repo.

Have config import named in package.json and  does not have any export in src/App.js

when run npm run start script, no error in terminal.

**Comment by ljharb** [comment]
In that repo, when i run `npx eslint .` (all the CRA nonsense isn't necessary to reproduce it), i indeed see no errors.

Enabling `import/default` should be catching this, but is not.

**Comment by ljharb** [comment]
ahhh - it's because the file is ambiguous, because it doesn't have any `import` or `export` statements, but it still might have a `module.exports` statement. The `named` rule has the same behavior.

I'm not sure there's anything we can do here. If you add `export {}` to the file, both rules error as expected. This is documented in both the `named` and `default` rule docs.
